src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -241,5 +280,7 @@ private ConnectionPool(ConnectionSettings cs)
 		readonly SemaphoreSlim m_sessionSemaphore;
 		readonly LinkedList<MySqlSession> m_sessions;
 		readonly ConnectionSettings m_connectionSettings;
+		readonly Dictionary<int, WeakReference<MySqlSession>> m_leasedSessions;
+		int m_lastId;
```

Comment:
I considered this, but even if you opened 10,000 connections/sec, it would take 4.97 days to wrap around, then you'd only have a problem (ID collision) if one of those connections had been kept open for five days. (Under more typical load, ~150 conn/s, it wouldn't wrap around for a year.)

The downside of `long` is that `Interlocked.Increment` has to be simulated on a 32-bit system, which is still a potential target platform for this code.